Compile failures since version 2.0.0

Since shared-memory-ring 2.0.0 my project doesn't compile anymore.

The solver tries to downgrade cstruct to version 2.3.1, although 2.4.1 exists as well and also satisfies nocryptos "cstruct < 3.0.0". Compiling against cstruct 2.3.1 fails:

```
$ opam install shared-memory-ring
The following actions will be performed:
  ∗  install shared-memory-ring 2.0.0

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[shared-memory-ring] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of shared-memory-ring failed at "jbuilder build -p shared-memory-ring -j 4".

#=== ERROR while installing shared-memory-ring.2.0.0 ==========================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p shared-memory-ring -j 4
# path         /home/sanders/.opam/4.04.1/build/shared-memory-ring.2.0.0
# compiler     4.04.1
# exit-code    1
# env-file     /home/sanders/.opam/4.04.1/build/shared-memory-ring.2.0.0/shared-memory-ring-7541-1872ec.env
# stdout-file  /home/sanders/.opam/4.04.1/build/shared-memory-ring.2.0.0/shared-memory-ring-7541-1872ec.out
# stderr-file  /home/sanders/.opam/4.04.1/build/shared-memory-ring.2.0.0/shared-memory-ring-7541-1872ec.err
### stderr ###
# [...]
#          ppx lib/xenstore_ring.pp.ml
#          ppx lib/ring.pp.ml
#     ocamldep lib/shared_memory_ring.depends.ocamldep-output
#       ocamlc lib/ring.{cmi,cmti}
#       ocamlc lib/xenstore_ring.{cmi,cmti}
#       ocamlc lib/console_ring.{cmi,cmti}
#       ocamlc lib/xenstore_ring.{cmo,cmt} (exit 2)
# (cd _build/default && /home/sanders/.opam/4.04.1/bin/ocamlc.opt -w -40 -g -bin-annot -I /home/sanders/.opam/4.04.1/lib/bytes -I /home/sanders/.opam/4.04.1/lib/cstruct -I /home/sanders/.opam/4.04.1/lib/ocaml -I /home/sanders/.opam/4.04.1/lib/ocplib-endian -I /home/sanders/.opam/4.04.1/lib/sexplib -I /home/sanders/.opam/4.04.1/lib/sexplib/0 -no-alias-deps -I lib -o lib/xenstore_ring.cmo -c -impl lib/xenstore_ring.pp.ml)
# File "lib/xenstore_ring.ml", line 25, characters 5-12:
# Error: Uninterpreted extension 'cstruct'.



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install shared-memory-ring 2.0.0
No changes have been performed
$ opam show -f version cstruct
2.3.1
```

If I pin cstruct to 2.4.1, shared-memory-ring 2.0.0 compiles, but there is a follow up error while compiling mirage-xen:

```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed shared-memory-ring.2.0.0
[ERROR] The compilation of mirage-xen failed at "make xen-build".
Processing  2/15: [mirage-xen: make xen-uninstall]
#=== ERROR while installing mirage-xen.3.0.0 ==================================#
# opam-version 1.2.2
# os           linux
# command      make xen-build
# path         /home/sanders/.opam/4.04.1/build/mirage-xen.3.0.0
# compiler     4.04.1
# exit-code    2
# env-file     /home/sanders/.opam/4.04.1/build/mirage-xen.3.0.0/mirage-xen-13742-a4abd5.env
# stdout-file  /home/sanders/.opam/4.04.1/build/mirage-xen.3.0.0/mirage-xen-13742-a4abd5.out
# stderr-file  /home/sanders/.opam/4.04.1/build/mirage-xen.3.0.0/mirage-xen-13742-a4abd5.err
### stdout ###
# cd xen && make
# make[1]: Entering directory `/home/sanders/.opam/4.04.1/build/mirage-xen.3.0.0/xen'
# ocaml setup.ml -configure 
# make[1]: Leaving directory `/home/sanders/.opam/4.04.1/build/mirage-xen.3.0.0/xen'
### stderr ###
# [...]
# ocamlfind: Package `shared-memory-ring.xenstore' not found
# W: Field 'pkg_shared_memory_ring_xenstore' is not set: Command ''/home/sanders/.opam/4.04.1/bin/ocamlfind' query -format %d shared-memory-ring.xenstore > '/tmp/oasis-de61d4.txt'' terminated with error code 2
# ocamlfind: Package `shared-memory-ring.console' not found
# W: Field 'pkg_shared_memory_ring_console' is not set: Command ''/home/sanders/.opam/4.04.1/bin/ocamlfind' query -format %d shared-memory-ring.console > '/tmp/oasis-db0693.txt'' terminated with error code 2
# E: Cannot find findlib package shared-memory-ring.console
# E: Cannot find findlib package shared-memory-ring.lwt
# E: Cannot find findlib package shared-memory-ring.xenstore
# E: Failure("3 configuration errors")
# make[1]: *** [setup.data] Error 1
# make: *** [xen-build] Error 2



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions were aborted
  ∗  install cohttp             0.20.1
  ∗  install conduit            0.15.3
  ∗  install mirage-bootvar-xen 0.4.0 
  ∗  install mirage-conduit     2.3.1 
  ∗  install mirage-entropy     0.4.0 
  ∗  install mirage-http        3.1.0 
  ∗  install mirage-net-xen     1.7.0 
  ∗  install mirage-os-shim     3.0.0 
  ∗  install nocrypto           0.5.4 
  ∗  install tcpip              3.1.3 
  ∗  install tls                0.8.0 
  ∗  install webmachine         0.3.2 
  ∗  install x509               0.5.3 
The following actions failed
  ∗  install mirage-xen 3.0.0
The following changes have been performed
  ∗  install shared-memory-ring 2.0.0
```

Pinning shared-memory-ring to 1.3.0 solves the issues.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,3 +1,7 @@
+## 2.0.1 (2017-06-07)
+* Don't force clients to link against `cstruct.ppx`: this is the support code
+  for the PPX rewriter itself, and brings in the `compiler-libs` package etc.
+
 ## 2.0.0 (2017-05-28)
 * Refactor the ocamlfind packages and opam packages into
   - shared-memory-ring: all the portable code including the -xen definitions

--- a/shared-memory-ring-lwt.opam
+++ b/shared-memory-ring-lwt.opam
@@ -15,7 +15,8 @@ build: [
 depends: [
   "ocamlfind" {build}
   "jbuilder"  {build & >="1.0+beta9"}
-  "cstruct" {>= "1.9.0"}
+  "cstruct" {>= "2.4.1" }
+  "ppx_cstruct" {build}
   "shared-memory-ring"
   "lwt"
   "mirage-profile"

--- a/shared-memory-ring.opam
+++ b/shared-memory-ring.opam
@@ -15,8 +15,8 @@ build: [
 depends: [
   "ocamlfind" {build}
   "jbuilder"  {build & >="1.0+beta9"}
-  "cstruct" {>= "1.9.0"}
-  "ppx_tools"
+  "cstruct" {>= "2.4.1"}
+  "ppx_cstruct" {build}
   "mirage-profile"
   "ounit" {test}
 ]

